spec/views/user_index_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+RSpec.describe 'User pages test index/show ', type: :feature do
+  before(:all) do
+    @first_user = User.create(name: 'Solomon', photo: 'https://avatars.githubusercontent.com/u/101924220?v=4',
+                              bio: 'test for User')
+    @post1 = Post.create(title: 'Rspec test 1', text: 'rspec test for post', user_id: @first_user)
+    @post2 = Post.create(title: 'Rspec test 2', text: 'rspec test for post', user_id: @first_user)
+    @post3 = Post.create(title: 'Rspec test 3', text: 'rspec test for post', user_id: @first_user)
+    @post4 = Post.create(title: 'Rspec test 4', text: 'rspec test for post', user_id: @first_user)
+  end
+
+  describe 'User index page test' do
+    it 'Should show the username ' do
+      visit users_path
+      expect(page).to have_content(@first_user.name)
+    end
+
+    it 'Should show the username of all users' do
+      visit users_path
+      expect(page).to have_content(@first_user.name)
+    end
+
+    it 'Should show the user photo for all users' do
+      visit users_path
+      expect(page).to have_css('img')
+    end
+
+    it 'Should show the number of posts for each user' do
+      visit users_path
+      expect(page).to have_content(@first_user.posts_counter)
+    end
+
+    it 'Should show the profile picture' do
+      visit users_path
+      expect(page).to have_selector("img[src='#{@first_user.photo}']")
+    end
+
+    it 'Should show button to see all posts ' do
+      visit user_path(@first_user)
+      expect(page).to have_link('See all posts')
+    end
```

Comment:
- [ ] I think this test should be in the User show page which is missing. Kindly create this and add all the required tests for this page as per the screenshots. Some of your tests in the user_index_spec should be in the user_show_spec. Kindly restructure your view spec files.
![Screen Shot 2023-02-15 at 3 22 25 AM](https://user-images.githubusercontent.com/62849872/219014175-0fe678e2-716f-430c-98a6-37b23e163c33.png)